firmware/library/L1_Drivers/onboardled.hpp

```diff
@@ -7,14 +7,96 @@
 #pragma once
 
 #include <cstdint>
+#include "L1_Drivers/gpio.hpp"
+#include "L2_Utilities/macros.hpp"
 
 class OnBoardLedInterface {
  public:
-           virtual void Initialize()                                 = 0;
-           virtual void LedOn(uint8_t led_number)                    = 0;
-           virtual void LedOff(uint8_t led_number)                   = 0;
-           virtual void LedSet(uint8_t led_number, bool on)          = 0;
-           virtual void LedToggle(uint8_t led_number)                = 0;
-           virtual void LedSetAll(uint8_t value)                     = 0;
-           virtual uint8_t LedGetValues(void)                        = 0;
+  enum LightState : bool
+  {
+  kOn = true,
+  kOff = false
+  };
+           virtual void Initialize()                                      = 0;
+           virtual void On(uint8_t led_number)                            = 0;
+           virtual void Off(uint8_t led_number)                           = 0;
+           virtual void Set(uint8_t led_number,
+              OnBoardLedInterface::LightState state)                      = 0;
+           virtual void Toggle(uint8_t led_number)                        = 0;
+           virtual void SetAll(uint8_t value)                             = 0;
```

Comment:
This could take the LightState enumeration as an input.

Reply:
My original vision of Toggle would only "flip" the state of the LED based on the given number. My original vision of SetAll() would take in an actual value. For example 0b0000'1010, which would make LED3 and LED1 on, but LED2 and LED0 off. I'm not sure how I could make that enumeration work in either case. I am open to suggestions on how to make this more clear for someone who will use this library.